non-wheel/httpcrtsrv/httpsrv.py

```diff
@@ -0,0 +1,444 @@
+        index_file = open("%s/example-project/htdocs/index.html" % path, "w+")
+        index_file.write(get_index_html('Example Project'))
+        index_file.close()
+        os.mkdir('%s/example-project/logs' % path)
+        os.symlink("%s/example-project/conf/%s.conf" % (path, name), "%s/conf/apache-sites/example-project.conf" % path)
+
+        create_with_sql(name)
+
+
+# Hopefully to go away in favor of create_with_qzui_api
+def create_with_sql(name):
+    # This connection is for the SQL way of creating an HTTP Server Instance.
+    conn = db2.connect()
+    cur = conn.cursor()
+
+    sys_cmd = "CPYF FROMFILE(QUSRSYS/QATMHINSTC) TOFILE(QUSRSYS/QATMHINSTC) " +\
+            "FROMMBR(APACHEDFT) TOMBR(%s) MBROPT(*REPLACE)" % name
+    crt_http_svr = "call qcmdexc('%s');" % sys_cmd
+    cur.execute(crt_http_svr)
+
+    # This is the part that does not work properly.
+    mod_info_query = '''
+create or replace alias QUSRSYS.QATMHINSTC_%s for QUSRSYS.QATMHINSTC(%s);
+update QUSRSYS.QATMHINSTC_%s
+set CHARFIELD = '-apache -d /www/%s -f conf/httpd.conf -AutoStartN';
+''' % (name, name, name, name)
+
+    cur.execute(mod_info_query)
+    conn.commit()
+
+    cur.close()
+    conn.close()
+
+
+# This is the part that needs a lot of help.
+# I (Josh) do not know the Python Toolkit for IBM i. I'm willing to learn.
+def create_with_qzui_api():
+    itool = iToolKit()
+    # itool.add(
+    #     iSrvPgm('qzuicrtins', 'QZUICRTINS', 'QzuiCreateInstance')
+    #         .addParm(iData('name', '10a', options.devenv))
+    #     # .addParm('idata', '')
+    # )
+    # itool.call(config.itransport)
+
+
+def main():
+    usage = '''
+    python3 httpsrv.py (-c|d|r) [--conf=<template>] [--name=<name>] [--port=<number>] [--newname=<name>]
+    python3 httpsrv.py -c --conf=zendsvr6 --name=test --port=10090
+    python3 httpsrv.py -r --name=test --newname=develop
+    python3 httpsrv.py -d --name=develop
+'''
+    p = optparse.OptionParser(usage)
+    p.add_option('-c', action='store_true', dest="create", help='Create HTTP Server Instance')
+    p.add_option('-d', action='store_true', dest="delete", help='Delete HTTP Server Instance')
+    p.add_option('-r', action='store_true', dest="rename", help='Rename HTTP Server Instance')
+    p.add_option('--conf', '-t', default="zendphp7",
+         help='Template to use for Apache Config. zendphp7|zendsvr6|apache [default: zendphp7]')
+    p.add_option('--name', '-n', help="Name of HTTP Server Instance")
+    p.add_option('--newname', '-e', help="New name for HTTP Server Instance")
+    p.add_option('--port', '-p', help="Default port for HTTP Server Instance")
+    options, arguments = p.parse_args()
+
+    if len([x for x in (options.create, options.delete, options.rename) if x is not None]) != 1:
+        p.error('options -c, -d, and -r cannot be used together.')
+
+    if options.create:
+        if len([x for x in (options.name, options.conf, options.port) if x is not None]) == 3:

```

Comment:
This can also be done with [subparsers](https://docs.python.org/3/library/argparse.html#argparse.ArgumentParser.add_subparsers), but you would have to change to using positional arguments instead of flags for `--create`, `--rename`, and `--delete`:

```
python3 httpsrv.py -c --conf=zendsvr6 --name=test --port=10090
python3 httpsrv.py -r --name=test --newname=develop
python3 httpsrv.py -d --name=develop
```
to
```
python3 httpsrv.py create --conf=zendsvr6 --name=test --port=10090
python3 httpsrv.py rename --name=test --newname=develop
python3 httpsrv.py delete --name=develop
```